core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -1,69 +1,279 @@
+          metadata.getClusteringOrder(ck) == Order.ASC ? Direction.ASC : Direction.DESC;
+      createTableWithOptions = createTableWithOptions.clusteringOrder(ck, direction);
+    }
+    // Add compaction strategy
+    CompactionStrategy compactionStrategy =
+        CompactionStrategy.valueOf(
+            options.getOrDefault(COMPACTION_STRATEGY, CompactionStrategy.STCS.toString()));
+
+    CompactionOptions<?> strategy;
+    switch (compactionStrategy) {
+      case LCS:
+        strategy = SchemaBuilder.leveledStrategy();
+        break;
+      case TWCS:
+        strategy = SchemaBuilder.timeWindowCompactionStrategy();
+        break;
+      default:
+        strategy = SchemaBuilder.sizedTieredStategy();
+    }
+    createTableWithOptions = createTableWithOptions.compactionOptions(strategy);
+
+    try {
+      clusterManager.getSession().execute(createTableWithOptions.getQueryString());
+    } catch (RuntimeException e) {
+      throw new ExecutionException(
+          String.format("creating the table %s.%s failed", fullKeyspace, table), e);
+    }
+  }
+
+  @VisibleForTesting
+  void createSecondaryIndex(String fullKeyspace, String table, Set<String> secondaryIndexNames)
+      throws ExecutionException {
+    for (String index : secondaryIndexNames) {
+      String indexName = String.format("%s_%s_%s", table, INDEX_NAME_PREFIX, index);
+      SchemaStatement createIndex =
+          SchemaBuilder.createIndex(indexName).onTable(fullKeyspace, table).andColumn(index);
+      try {
+        clusterManager.getSession().execute(createIndex.getQueryString());
+      } catch (RuntimeException e) {
+        throw new ExecutionException(
+            String.format(
+                "creating the secondary index for %s.%s.%s failed", fullKeyspace, table, index),
+            e);
+      }
+    }
   }
 
   @Override
   public void close() {
     clusterManager.close();
   }
+
+  /**
+   * Returns the equivalent {@link com.datastax.driver.core.DataType}
+   *
+   * @return the equivalent {@link com.datastax.driver.core.DataType} to the Scalar DB data type
+   */
+  private com.datastax.driver.core.DataType toCassandraDataType(DataType dataType) {
+    switch (dataType) {
+      case BOOLEAN:
+        return com.datastax.driver.core.DataType.cboolean();
+      case INT:
+        return com.datastax.driver.core.DataType.cint();
+      case BIGINT:
+        return com.datastax.driver.core.DataType.bigint();
+      case FLOAT:
+        return com.datastax.driver.core.DataType.cfloat();
+      case DOUBLE:
+        return com.datastax.driver.core.DataType.cdouble();
+      case TEXT:
+        return com.datastax.driver.core.DataType.text();
+      case BLOB:
+        return com.datastax.driver.core.DataType.blob();
+      default:
+        throw new UnsupportedOperationException(String.format("%s is not yet supported", dataType));
+    }
+  }
+
+  enum CompactionStrategy {
+    STCS,
+    LCS,
+    TWCS
+  }
+
+  enum NetworkStrategy {
```

Comment:
```suggestion
  enum ReplicationStrategy {
```
I think this is how it's usually called.
https://docs.datastax.com/en/cassandra-oss/3.0/cassandra/architecture/archDataDistributeReplication.html